lightning/src/ln/channelmanager.rs

```diff
@@ -3522,6 +3529,54 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 		}
 	}
 
+	/// Gets the set of txids which should be monitored for their confirmation state.
+	///
+	/// If you're providing information about reorganizations via [`transaction_unconfirmed`], this
+	/// is the set of transactions which you may been to call [`transaction_unconfirmed`] for.
+	///
+	/// Note that this is NOT the set of transactions which must be included in calls to
+	/// [`transactions_confirmed`] if they are confirmed, but a small subset of it.
+	///
+	/// [`transactions_confirmed`]: Self::transactions_confirmed
+	/// [`transaction_unconfirmed`]: Self::transaction_unconfirmed
+	pub fn get_relevant_txids(&self) -> Vec<Txid> {
+		let channel_state = self.channel_state.lock().unwrap();
+		let mut res = Vec::with_capacity(channel_state.short_to_id.len());
+		for chan in channel_state.by_id.values() {
+			if let Some(funding_txo) = chan.get_funding_txo() {
+				res.push(funding_txo.txid);
+			}
+		}
+		res
+	}
+
+	/// Marks a transaction as having been reorganized out of the blockchain.
```

Comment:
I'm confused because this indicates the transaction was def reorged out, but the docs for `channel.transaction_unconfirmed` say "...may also indicate a harmless reorganization of a block or two."? Seems to be a contradiction?

Reply:
Its referring to the case where we didn't send a funding_locked yet. I rephrased it.